### DISCUSSION TITLE
Transportation theme access restrictions "during" as STRUCT

### DISCUSSION THREAD

**Discussion by BruceHarold** [OvertureMaps/data#296] [Discussion]
At 2024-06-13-beta.1 the **during** column in access_restrictions is a varchar with uncontrolled data in it (my observation).
For example 'Mo-Fr08:45-09:3015:00-15:45'. for segment id 084194adffffffff046bf1ffa89acc2c.
To unpack this into very useful time intervals can it me made a struct, like this? (haven't checked for typos)
{"Mon":[["8:45","9:30"],["15:00","15:45"]],"Tue":[["8:45","9:30"],["15:00","15:45"]],"Wed":[["8:45","9:30"],["15:00","15:45"]],"Thu":[["8:45","9:30"],["15:00","15:45"]],"Fri":[["8:45","9:30"],["15:00","15:45"]]}
Time data is high value.  Thanks.

**Comment by brad-richardson** [comment]
We follow the OSM opening hours specification, although that fact should be better documented: https://github.com/OvertureMaps/schema/blob/main/schema/defs.yaml#L153C11-L153C78 It's probably worth considering a new structure for this data as you've proposed, but I would say it's currently a low priority for the team.

Just FYI - We do some processing/normalizing internally to fix some of the values but there are known issues. I would highly recommend using a library like https://github.com/remi-dupre/opening-hours-rs/tree/master/python to handle this as the rules are fairly complex.

**Reply by BruceHarold** [reply]
Thanks Brad.  I'm heading to a normalized view of the data so it is easily queryable, say a table you can join on id with day, opens & closes columns.  The OSM grammar makes this (very) hard.  I see wrinkles like named times ("sunrise") which could be handled separately.
I'll stay subscribed to this and we'll see who else picks up on this.  I'm planning on a blog on the transportation theme data for Esri users.

**Comment by BruceHarold** [comment]
An idea to handle OSM's dawn|sunrise|sunset|dusk events would be to calculate those for the median date of each Overture release at the latitude of the feature.
